core/server/worker/src/main/java/alluxio/worker/netty/DataServerShortCircuitWriteHandler.java

```diff
@@ -74,6 +77,13 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable throwable) {
     ctx.close();
   }
 
+  @Override
+  public void channelUnregistered(ChannelHandlerContext ctx) {
+    if (mSessionId > 0) {
+      mBlockWorker.cleanupSession(mSessionId);
+    }
```

Comment:
Do we want to fire channel unregistered here? Also why do we need to check sessionId > 0?